src/adapter/src/catalog/builtin.rs

```diff
@@ -2140,6 +2140,24 @@ FROM mz_internal.mz_dataflow_operator_dataflows_per_worker
 WHERE worker_id = 0",
 };
 
+pub const MZ_OBJECT_DEPENDENCIES_TRANSITIVE: BuiltinView = BuiltinView {
+    name: "mz_object_dependencies_transitive",
+    schema: MZ_INTERNAL_SCHEMA,
+    sql: "CREATE VIEW mz_internal.mz_object_dependencies_transitive AS
+WITH MUTUALLY RECURSIVE
+  base(src_id text, tgt_id text) AS (
+    SELECT object_id, referenced_object_id FROM mz_internal.mz_object_dependencies
+  ),
+  reach(src_id text, tgt_id text) AS (
+    SELECT src_id, tgt_id FROM base
+    UNION ALL -- (TODO use UNION once #19817 is fixed)
+    SELECT src_id, tgt_id FROM reach
+    UNION
+    SELECT r1.src_id, r2.tgt_id FROM reach r1 JOIN reach r2 ON r1.tgt_id = r2.src_id
+  )
+SELECT src_id, tgt_id FROM reach;",
```

Comment:
As per the catalog style guide (https://github.com/MaterializeInc/materialize/blob/main/doc/developer/style.md#system-catalog-style), we should use full words for column names, i.e. `source_id` and `target_id` (`id` is an allowed abbreviation).

Reply:
Changed the schema to `id text, referenced_object_id text` (identical with `mz_object_dependencies`).